src/radical/pilot/utils/prof_utils.py

```diff
@@ -961,6 +961,22 @@ def get_consumed_resources(session, rtype='cpu', tdurations=None):
             log.debug('    %10.2f  %-20s  %-15s  %-15s  %-15s  %-15s  %s',
                            ts[0],  ts[1], ts[2], ts[3], ts[4], ts[5], ts[6])
 
+        p_min = None
+        p_max = None
+
+        try   : p_min = pilot.timestamps(event={1: 'bootstrap_0_start'})[0]
+        except: pass
+
+        try   : p_max = pilot.timestamps(event={1: 'bootstrap_0_stop'})[0]
+        except: pass
+
+        # fallback for missing bootstrap events
+        if p_min is None: p_min = pilot.timestamps(state='PMGR_ACTIVE')
+        if p_max is None: p_max = pilot.events[-1][ru.TIME]
+
+        assert(p_min is not None)
+        assert(p_max is not None)
+
         p_min = pt(event=PILOT_DURATIONS['consume']['idle'][0]) [0]
         p_max = pt(event=PILOT_DURATIONS['consume']['idle'][1])[-1]
```

Comment:
section with analytics having failure with `p_max` while trying to get event
```
{ru.EVENT: 'cmd'              ,
 ru.MSG  : 'cancel_pilot'     }
```

Shouldn't that last two lines (L980,981) be part of the code block above?

p.s. also still cannot figure out why `cancel_pilot` could be missed, as an example on Summit I had that event
```
./pilot.0000/agent.0.prof:1649780740.4137278,cmd,agent.0,Thread-9,pilot.0000,,cancel_pilot : []
```

Reply:
> also still cannot figure out why cancel_pilot could be missed,

the event will be missing if the pilot does not cancel on it's own fast enough and the pmgr kills the pilot job instead.